src/casl/casl-ability.factory.ts

```diff
@@ -54,40 +54,47 @@ export type AppAbility = MongoAbility<PossibleAbilities, Conditions>;
 
 @Injectable()
 export class CaslAbilityFactory {
-  createForUser(user: JWTUser) {
+  endpointAccess(user: JWTUser, subject: string) {
+    switch (subject) {
+      case "datasets":
+        return this.datasetEndpointAccess(user);
+      case "elastic-search":
+        return this.elasticSearchEndpointAccess(user);
+      case "jobs":
+        return this.jobsEndpointAccess(user);
+      case "instruments":
+        return this.instrumentEndpointAccess(user);
+      case "logbooks":
+        return this.logbookEndpointAccess(user);
+      case "origdatablocks":
+        return this.origDatablockEndpointAccess(user);
+      case "policies":
+        return this.policyEndpointAccess(user);
+      case "proposals":
+        return this.proposalsEndpointAccess(user);
+      case "publisheddata":
+        return this.publishedDataEndpointAccess(user);
+      case "samples":
+        return this.samplesEndpointAccess(user);
+      case "users":
+      case "useridentities":
+        return this.userEndpointAccess(user);
```

Comment:
So yes, regarding this and the previous comment about the way the endpoint is extracted, I was also worried that it might break something in future. The alternative would be to unite the endpoint functions back into one, eliminating the switch statement because it is then based on the user only again. This is then not as granular as now, but at the same time the endpoint functions are also not that long/complex.

Reply:
@sofyalaski what do you think is the best to move forward?

Reply:
I added a new commit, it now changes the decorator for @CheckPolices to accept two arguments, the first being the keyword, according to which the function will be sent to the correct endpoint function. What do you think?